test/suites/integration/bsp/multi-volunteer/multi-volunteer.test-2.ts

```diff
@@ -0,0 +1,86 @@
+import { strictEqual } from "node:assert";
+import { isDeepStrictEqual } from "node:util";
+import invariant from "tiny-invariant";
+import {
+  bspThreeKey,
+  bspThreeSeed,
+  bspTwoKey,
+  bspTwoSeed,
+  describeBspNet,
+  type EnrichedBspApi,
+  ShConsts,
+  shUser
+} from "../../../../util";
+
+describeBspNet("BSPNet: Mulitple BSP Volunteering - 2", ({ before, it, createUserApi }) => {
+  let api: EnrichedBspApi;
+
+  before(async () => {
+    api = await createUserApi();
+  });
+
+  it("multiple BSPs race to volunteer ", async () => {
+    // Replicate to 1 BSPs, 1 block to maxthreshold (i.e. instant acceptance)
+    await api.sealBlock(api.tx.sudo.sudo(api.tx.fileSystem.setGlobalParameters(1, 1)));
+
+    await api.docker.onboardBsp({
+      bspSigner: bspTwoKey,
+      name: "sh-bsp-two",
+      bspKeySeed: bspTwoSeed,
+      bspId: ShConsts.BSP_TWO_ID,
+      additionalArgs: ["--keystore-path=/keystore/bsp-two"]
+    });
+
+    await api.docker.onboardBsp({
+      bspSigner: bspThreeKey,
+      name: "sh-bsp-three",
+      bspKeySeed: bspThreeSeed,
+      bspId: ShConsts.BSP_THREE_ID,
+      additionalArgs: ["--keystore-path=/keystore/bsp-three"]
+    });
+
+    const bucketEvent = await api.file.newBucket("multi-bsp-single-req");
+    const newBucketEventDataBlob =
+      api.events.fileSystem.NewBucket.is(bucketEvent) && bucketEvent.data;
+
+    invariant(newBucketEventDataBlob, "Event doesn't match Type");
+
+    const fileMetadata = await api.rpc.storagehubclient.loadFileInStorage(
+      "res/adolphus.jpg",
+      "cat/adolphus.jpg",
+      shUser.address,
+      newBucketEventDataBlob.bucketId
+    );
+
+    const signedExt = await api.tx.fileSystem
+      .issueStorageRequest(
+        newBucketEventDataBlob.bucketId,
+        "cat/adolphus.jpg",
+        fileMetadata.fingerprint,
+        fileMetadata.file_size,
+        ShConsts.DUMMY_MSP_ID,
+        [ShConsts.NODE_INFOS.user.expectedPeerId]
+      )
+      .signAsync(shUser);
+
+    await api.sealBlock(signedExt);
+    await api.wait.bspVolunteer();
+    await api.wait.bspStored();
+
+    const matchedEvents = await api.assert.eventMany("system", "ExtrinsicFailed");
```

Comment:
Might be worth commenting what's going on behind the scenes here, as it's kinda hidden behind the `wait`s. So from my understanding, in `api.wait.bspVolunteer()` you wait for at least one volunteer (but there will actually be 3), and then in `api.wait.bspStored()` you wait for the FIRST confirmation. Then there should be 2 failing ones.